### PR TITLE
DEP cleanup linting debugging

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Pushed the branch to main repo
* [x] CI passed on the branch

### CODE REVIEW

**Comment by isuruf** [comment]
Here's a log I found yesterday that gives some info,
```
2020-04-17T22:53:08.933302+00:00 app[web.1]: WARNING:tornado.access:404 POST /conda-forge-command/org-hook (140.82.115.243) 1.45ms
2020-04-17T22:53:08.986263+00:00 app[web.1]: 
2020-04-17T22:53:08.986355+00:00 app[web.1]: ===================================================
2020-04-17T22:53:08.986471+00:00 app[web.1]: linting: conda-forge/libcxx-feedstock
2020-04-17T22:53:08.986512+00:00 app[web.1]: ===================================================
2020-04-17T22:53:11.061712+00:00 app[web.1]: ERROR:ERROR: expected a single document in the stream
2020-04-17T22:53:11.061731+00:00 app[web.1]: in "<unicode string>", line 65, column 9:
2020-04-17T22:53:11.061732+00:00 app[web.1]: - include/c++
2020-04-17T22:53:11.061732+00:00 app[web.1]: ^ (line: 65)
2020-04-17T22:53:11.061732+00:00 app[web.1]: but found another document
2020-04-17T22:53:11.061733+00:00 app[web.1]: in "<unicode string>", line 67, column 9:
2020-04-17T22:53:11.061733+00:00 app[web.1]: - lib/libc++.*
2020-04-17T22:53:11.061820+00:00 app[web.1]: ^ (line: 67)
2020-04-17T22:53:11.061880+00:00 app[web.1]: string index out of range
2020-04-17T22:53:11.814188+00:00 app[web.1]: ERROR: 1
2020-04-17T22:53:11.819112+00:00 app[web.1]: ERROR: while parsing a block mapping
2020-04-17T22:53:11.819114+00:00 app[web.1]: in "/tmp/tmp0zgfvrpi_recipe/recipe/../conda-forge.yml", line 5, column 3
2020-04-17T22:53:11.819114+00:00 app[web.1]: expected <block end>, but found '{'
2020-04-17T22:53:11.819115+00:00 app[web.1]: in "/tmp/tmp0zgfvrpi_recipe/recipe/../conda-forge.yml", line 6, column 11
2020-04-17T22:53:13.421468+00:00 app[web.1]: 
2020-04-17T22:53:13.421568+00:00 app[web.1]: GitHub API Rate Limit Info:
2020-04-17T22:53:13.638229+00:00 app[web.1]: ------------------------------------------------
2020-04-17T22:53:13.638236+00:00 app[web.1]: conda-forge-linter - remaining 4482 out of 5000.
2020-04-17T22:53:13.638240+00:00 app[web.1]: Will reset in 0:20:55.361849.
2020-04-17T22:53:13.639688+00:00 app[web.1]:
```

Looks like `conda-forge.yml` gets corrupted.

**Comment by beckermr** [comment]
That is useful! Thanks @isuruf!

File: conda_forge_webservices/utils.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+import uuid
 from contextlib import contextmanager
 
 
 @contextmanager
 def tmp_directory():
-    tmp_dir = tempfile.mkdtemp('_recipe')
+    # I am pretty sure the uuid here does nothjing, but lets make sure the
+    # tmpdir is always unique
+    tmp_dir = tempfile.mkdtemp('_recipe_h' + uuid.uuid4().hex[0:6])
```
Comment (isuruf) [inline conda_forge_webservices/utils.py:12]: It is always unique. The name here is a suffix to an uuid like string

Comment (beckermr) [reply conda_forge_webservices/utils.py:12]: This is what I found locally when I tested on my mac too.

**Comment by beckermr** [comment]
Looking at this traceback, it looks like the yaml parser switched documents midparsing. Maybe the linting code in smithy is not thread safe.

**Comment by beckermr** [comment]
Here might be it 

https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/lint_recipe.py#L550

We reuse the ruamel.yaml parser in conda smithy and this may not be thread safe.

**Comment by isuruf** [comment]
Yes, I came to the same conclusion. https://github.com/conda-forge/conda-forge-webservices/pull/361 is a workaround until we fix conda-smithy.

**Comment by beckermr** [comment]
Cleaning out the debugging stuff.

**Pull request merged**